### DISCUSSION TITLE
Make `helm-source-async` split the output by `\0` instead of `\n`

### DISCUSSION THREAD

**Discussion by NightMachinery** [emacs-helm/helm#2450] [Q&A] [answered]
I have a process that outputs multiline candidates separated by null characters `\0`. How do I make an `helm-source-async` that splits on `\0`?

**Comment by thierryvolpiatto** [comment, marked as answer]
NightMachinary ***@***.***> writes:

> I have a process that outputs multiline candidates separated by null characters \0. How do I make an helm-source-async that splits on \0?

In `helm-output-filter--process-source` helm split output with "\n", try
to modify it to your needs, if it works as expected I could make a new
var to handle this so that you can let-bound it.

-- 
Thierry

**Reply by NightMachinery** [reply]
This worked:
```elisp
(defvar helm-process-output-split-string-re "\n")
(defun helm-output-filter--process-source (process output-string source limit)
  (cl-dolist (candidate
              (helm-transform-candidates
               (helm-output-filter--collect-candidates
                (split-string output-string helm-process-output-split-string-re)
                (assq 'incomplete-line source))
               source t))
    (setq candidate
          (helm--maybe-process-filter-one-by-one-candidate candidate source))
    (if (assq 'multiline source)
        (let ((start (point)))
          (helm-insert-candidate-separator)
          (helm-insert-match candidate 'insert-before-markers
                             (1+ (cdr (assq 'item-count source)))
                             source)
          (put-text-property start (point) 'helm-multiline t))
      (helm-insert-match candidate 'insert-before-markers
                         (1+ (cdr (assq 'item-count source)))
                         source))
    (cl-incf (cdr (assq 'item-count source)))
    (when (>= (assoc-default 'item-count source) limit)
      (helm-kill-async-process process)
      (helm-log-run-hook 'helm-async-outer-limit-hook)
      (cl-return))))
```

I also needed to set `:multiline t` in my `source`.

Should I send a PR?

PS: Now that multiline candidates work, I can't find what command scrolls the helm window itself. I tried `(define-key helm-map (kbd "S-<up>") #'scroll-down-command)`, but this tries to scroll the minibuffer, not the helm window.

**Comment by thierryvolpiatto** [comment]
Now implemented, the var name is `helm-process-output-split-string-separator`, for scrolling you can now use C-v and M-v, they obey to `helm-scroll-amount`.
Thanks and sorry for delay, your last post was not mailed to me for some reasons and I then forget about it.
